converted current debugging statements in LLVM codegen to jit-logging statements #48771

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49121 added macros in jit logging to check whether loggings are enabled; replaced similar checks in LLVM codegen with such macros
* **#49040 converted current debugging statements in LLVM codegen to jit-logging statements #48771**



Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

huiguoo committed:
converted current debugging statements in LLVM codegen to jit-logging statements #48771

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/llvm_codegen.cpp:
```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
-  TM_->addPassesToEmitFile(
+  // print graph debug info after optimization
+  asmBuffer.set_size(0);
+  if (is_enabled(__FILE__, ::torch::jit::JitLoggingLevels::GRAPH_DEBUG)) {
```
Nit: might be a good idea to wrap it with a macro and make it commonly available (if you decide to do that, please put that into a separate PR).

huiguoo replied:
Sounds good. Will go ahead and add it.

huiguoo replied:
To confirm, are you suggesting adding a macro in `jit/jit_log.h` and making it available to all code that includes this header file? Using this macro, we can disable some computation required for loggings when logging is not enabled.

ZolotukhinM replied:
Yes, exactly that.

huiguoo replied:
Addressed in PR #49121.

ZolotukhinM commented on torch/csrc/jit/tensorexpr/llvm_codegen.cpp:
```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
-  // print graph debug info.
-  std::string fnstr;
-  llvm::raw_string_ostream FS(fnstr);
-  fn_->print(FS);
```
This print has disappeared, is it duplicating what's printed in the module?

huiguoo replied:
Yes, it's a duplicate. `module_->print()` prints the graph info generated by `fn_->print()` plus info associated with the module; it should be more helpful.

ZolotukhinM commented on torch/csrc/jit/tensorexpr/llvm_codegen.cpp:
```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
+  if (is_enabled(__FILE__, ::torch::jit::JitLoggingLevels::GRAPH_DEBUG)) {
+    module_->print(asmStream, nullptr);
+  }
+  GRAPH_DEBUG(asmStream.str().str(), "\n");
```
Nit: maybe add a header "LLVM module before optimizations" or something like that?

ZolotukhinM reviewed (approved):
Awesome!

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"


Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"


Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"


Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"




Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"




Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"




Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]

huiguoo committed:
Update on "converted current debugging statements in LLVM codegen to jit-logging statements #48771"




Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

[ghstack-poisoned]
